app/src/main/java/net/skyscanner/backpack/demo/compose/AppSearchModalStory.kt

```diff
@@ -117,6 +119,15 @@ internal fun DefaultAppSearchModalSample(
             onClose = { showModal.value = false },
             onInputChanged = { destination.value = it },
             clearAction = BpkClearAction(stringResource(id = R.string.text_field_clear_action_description)) { destination.value = "" },
+            behaviouralCallback = object : BehaviouralCallback {
+                override fun onDrawn(element: Any) {
+                    Log.i("AppSearchModalStory", "onDrawn $element")
```

Comment:
We can delete these log lines if needed- I wasn't sure whether to keep them in for the example or not

Reply:
yeah no harm doing so. i think having some tests for this would be important, though - for the most part we just use snapshot tests in backpack, but these call for some ui tests (there's already a few for some components)

Reply:
Good point- I'll just wait for a couple more reviews/iOS alignment before writing tests and docs, just in case the approach changes

Reply:
@marianeum I'm getting back to this now 😀 which class(es) would you recommend adding the tests for? For the high-level `BpkAppSearchModal`, which just passes the callback through, or the `BpkSectionItem` which actually contains the `registerForBehaviouralEvents` call?

Reply:
basically testing that the callback gets called at the right place with the right item

Reply:
@marianeum I've added one test now, does this match what you'd expect? If so I can cover the other test cases and add documentation

Reply:
yep that makes sense, thank you!